ParquetClassLibrary/AssemblyInfo.cs

```diff
@@ -11,94 +11,177 @@
+        /// Valid identifiers may be positive or negative.  By convention, negative IDs indicate test Items.
+        /// </summary>
+        public static readonly Range<EntityID> ItemIDs;
+        #endregion
+
+        #region Initialization
+        /// <summary>
+        /// Initializes the <see cref="Range{EntityID}"/>s defined in <see cref="AssemblyInfo"/>.
+        /// This supports defining ItemIDs in terms of the other Ranges.
+        /// </summary>
+        static AssemblyInfo()
+        {
+            // By convention, the first EntityID in each Range is a multiple of this number.
+            // An exception is made for Critter, Character, and Player as they are treated as sub-ranges.
+            var TargetMultiple = 10000;
+
+            FloorIDs = new Range<EntityID>(10000, 19000);
+            BlockIDs = new Range<EntityID>(20000, 29000);
+            FurnishingIDs = new Range<EntityID>(30000, 39000);
+            CollectibleIDs = new Range<EntityID>(40000, 49000);
+            CritterIDs = new Range<EntityID>(50000, 52900);
+            CharacterIDs = new Range<EntityID>(53000, 55900);
+            PlayerIDs = new Range<EntityID>(56000, 58900);
+            RoomRecipeIDs = new Range<EntityID>(60000, 69000);
+            CraftingRecipeIDs = new Range<EntityID>(70000, 79000);
+            QuestIDs = new Range<EntityID>(80000, 89000);
+
+            // The largest Range.Maximum defined in AssemblyInfo, excluding ItemIDs.
+            int MaximumIDNotCountingItems = typeof(AssemblyInfo).GetFields()
+                .Where(fieldInfo => fieldInfo.FieldType.IsGenericType
+                    && fieldInfo.FieldType == typeof(Range<EntityID>)
+                    && fieldInfo.Name != nameof(ItemIDs))
+                .Select(fieldInfo => fieldInfo.GetValue(null))
+                .Cast<Range<EntityID>>()
+                .Select(range => range.Maximum)
+                .Max();
+
+            // Since ItemIDs is being defined at runtime, its Range.Minimum must be chosen well above existing maxima.
+            var ItemLowerBound = TargetMultiple * ((MaximumIDNotCountingItems + (TargetMultiple - 1)) / TargetMultiple);
+
+            // The largest Range.Maximum of any paruet IDs.
+            int MaximumParquetID = ParquetIDs
+                .Select(range => range.Maximum)
+                .Max();
+
+            // The smallest Range.Minimum of any paruet IDs.
+            int MinimumParquetID = ParquetIDs
+                .Select(range => range.Minimum)
+                .Min();
+
+            // Since it is possible for every parquet to have a corresponding item, this range must be at least
+            // as large as all four parquet ranges put together.  Therefore, the Range.Maximum is twice the combined
+            // ranges of all parquets.
+            var ItemUpperBound = ItemLowerBound + 2 * (TargetMultiple / 10 + MaximumParquetID - MinimumParquetID);
+
+            ItemIDs = new Range<EntityID>(ItemLowerBound, ItemUpperBound);
```

Comment:
If ItemIDs' range is procedurally generated (albeit with a reasonable and predictable set of rules), would it be hard for someone making a block to provide an ItemID?

Reply:
Thanks for the convo about this, I am going to look at other APIs to see how this is handled.